src/v/redpanda/admin_server.cc

```diff
@@ -342,14 +346,16 @@ static json::validator make_set_replicas_validator() {
  * as an empty request body causes a redpanda crash via a rapidjson
  * assertion when trying to GetObject on the resulting document.
  */
-static json::Document parse_json_body(ss::http::request const& req) {
+ss::future<json::Document> parse_json_body(ss::http::request* req) {
     json::Document doc;
-    doc.Parse(req.content.data());
-    if (doc.Parse(req.content.data()).HasParseError()) {
```

Comment:
I may not fully understand rapidjson, but I don't think we need to call `Parse` twice.